File: frontend/web/controllers/window_controller.rb
# frozen_string_literal: true

class WindowController
  attr_accessor :hidden_windows

  def initialize
    @hidden_windows = []
  end

  # اخفاء نافذة
  def hide(window)
    @hidden_windows << window
    window.visible = false
  end

  # اظهار نافذة
  def show(window)
    window.visible = true
    @hidden_windows.delete(window)
  end

  # تكبير / تصغير
  def toggle_maximize(window)
    if window.maximized
      window.width = 600
      window.height = 600
      window.maximized = false
    else
      window.width = 1000
      window.height = 1000
      window.maximized = true
    end
  end

  # اغلاق نهائي
  def close(window)
    # إزالة النافذة من النظام
    # في التطبيق الفعلي يجب إزالة العنصر من DOM أو من قاعدة البيانات حسب التصميم
  end
end

File: frontend/web/controllers/window_controller.js
// window_controller.js
import { Controller } from "@hotwired/stimulus"

export default class extends Controller {
  static targets = ["window", "hiddenContainer"]

  connect() {
    console.log("Unified Responsive Window controller connected")
  }

  close(event) {
    const win = event.currentTarget.closest(".window")
    win.remove()
  }

  hide(event) {
    const win = event.currentTarget.closest(".window")
    const hiddenContainer = this.hiddenContainerTarget
    const icon = document.createElement("div")
    icon.className = "hidden-window"
    icon.dataset.target = win.dataset.id
    icon.textContent = win.dataset.icon || "📄"
    hiddenContainer.appendChild(icon)
    win.style.display = "none"

    icon.addEventListener("click", () => {
      win.style.display = "flex"
      icon.remove()
    })
  }

  toggleMaximize(event) {
    const win = event.currentTarget.closest(".window")
    if(this.isMobile()) return
    if (win.dataset.maximized === "true") {
      win.style.width = this.defaultWidth()
      win.style.height = this.defaultHeight()
      win.dataset.maximized = "false"
    } else {
      win.style.width = "100%"
      win.style.height = "100%"
      win.dataset.maximized = "true"
    }
  }

  isMobile() {
    return window.innerWidth <= 640
  }

  defaultWidth() {
    if(this.isMobile()) return "100vw"
    if(window.innerWidth <= 1024) return "80vw"
    return "600px"
  }

  defaultHeight() {
    if(this.isMobile()) return "100vh"
    if(window.innerWidth <= 1024) return "60vh"
    return "300px"
  }
}
